tests/src/core/testqgsexpression.cpp

```diff
@@ -1295,6 +1295,7 @@ class TestQgsExpression: public QObject
       QTest::newRow( "m_min line M NaN" ) << "m_min(make_line(geom_from_wkt('PointZM (0 0 0 nan)'),geom_from_wkt('PointZM (1 1 1 2)')))" << false << QVariant( );
       QTest::newRow( "m_min line" ) << "m_min(make_line(make_point_m(0,0,1),make_point_m(-1,-1,2),make_point_m(-2,-2,0)))" << false << QVariant( 0.0 );
       QTest::newRow( "main angle polygon" ) << "round(main_angle( geom_from_wkt('POLYGON((0 0,2 9,9 2,0 0))')))" << false << QVariant( 77 );
+      QTest::newRow( "main angle polygon edge case" ) << "round(main_angle( geom_from_wkt('POLYGON((353542.63843526 378974.92373469, 353544.95808017 378975.73690545, 353545.27173175 378974.84218528, 353542.95208684 378974.02901451, 353542.63843526 378974.92373469))')))" << false << QVariant( 71 );
```

Comment:
Arguably it would be better to also test QgsGeometry::orientedMinimumBoundingBox directly (so you don't only test for the angle, which doesn't necessarily mean width/height are right too).

It could easily be added here :
https://github.com/qgis/QGIS/blob/45c5a91d5f17126293f6addb86a412256bcd6cba/tests/src/core/geometry/testqgsgeometry.cpp#L2158

But IMO even without this it's already a valuable bugfix and already partially covered by test, so could also be merged as is if you don't manage to add that before next release

Reply:
Honestly, I'm not even sure how the bug occurs. When reprojecting the data it fails to bug. This seems to be a very edge case that is being fixed. I just assume that if the test is fine, so should the rest. I can look into it if needed.

Reply:
Thanks for the review also :)

Reply:
Maybe it would be nice to construct a test case with small integer numbers instead? That might make it more clear.

Reply:
@kannes if you have any way to reproduce the bug with integer coordinates I'll take your test, otherwise I won't waste a second more of my time on fiddling with this.

Reply:
You are right! Sorry, that comment of mine was unnecessary without providing a test case myself. Perfect is the enemy of good and I don't even know enough about QGIS' development and testing protocol. Thank you so much for working on this!

Reply:
@kannes Didn't mean to sound harsh, I just tried many things and this bug is quite hard to replicate, just moving the coordinates, changing projection or anything just reset something and make it behaves 'ok'. I tried many things and I just gave up.